pkg/ciliumenvoyconfig/cec_resource_parser.go

```diff
@@ -127,18 +147,17 @@ func (r *cecResourceParser) parseResources(cecNamespace string, cecName string,
 				listener.EnableReusePort = &wrapperspb.BoolValue{Value: false}
 			}
 
-			// Figure out if this is an internal listener
-			isInternalListener := listener.GetInternalListener() != nil
-
-			// Only inject Cilium filters if Cilium allocates listener address
-			injectCiliumFilters := listener.GetAddress() == nil && !isInternalListener
+			// Only inject Cilium filters if all of the following conditions are fulfilled
+			// * Cilium allocates listener address or it's a listener for a L7 loadbalancer
+			// * It's not an internal listener
+			injectCiliumFilters := (listener.GetAddress() == nil || isL7LB) && listener.GetInternalListener() == nil
```

Comment:
for other reviewer, isL7LB is added in this line :). Took me sometime to find out the change.